src/cpp/ext/otel/otel_plugin.cc

```diff
@@ -1007,4 +1037,9 @@ absl::Status OpenTelemetryPluginBuilder::BuildAndRegisterGlobal() {
   return impl_->BuildAndRegisterGlobal();
 }
 
+absl::StatusOr<std::shared_ptr<grpc::OpenTelemetryPlugin>>
+OpenTelemetryPluginBuilder::Build() {
+  return impl_->Build();
```

Comment:
Not directly related to this PR, but is there anything left stopping us from moving the functionality of `OpenTelemetryPluginBuilderImpl` directly into `OpenTelemetryPluginBuilder`, so that we can stop having this level of indirection?